src/middleware.rs

```diff
@@ -0,0 +1,52 @@
+use axum::{
+    body::Body,
+    extract::Request,
+    http::StatusCode,
+    middleware::Next,
+    response::{IntoResponse, Response},
+};
+use core::str;
+use http_body_util::BodyExt;
+use log::{log_enabled, trace, Level};
+
+pub async fn body_logger_middleware(
+    request: Request,
+    next: Next,
+) -> Result<impl IntoResponse, Response> {
+    if !log_enabled!(target: "live_compositor::log_request_body", Level::Trace) {
+        return Ok(next.run(request).await);
+    }
+    let request = buffer_request_body(request).await?;
+    let response = next.run(request).await;
+    let response = buffer_response_body(response).await?;
+
+    Ok(response)
+}
+
+async fn buffer_request_body(request: Request) -> Result<Request, Response> {
+    let (parts, body) = request.into_parts();
+
+    let bytes = body
+        .collect()
+        .await
+        .map_err(|err| (StatusCode::INTERNAL_SERVER_ERROR, err.to_string()).into_response())?
+        .to_bytes();
+
+    trace!(target: "live_compositor::log_request_body", "Request body: {:?}", str::from_utf8(&bytes).unwrap());
```

Comment:
- print other parts of the response too, like route or method, for response status code
- do not unwrap

Also, it would be great to try parsing requests as JSON values to remove the escaped characters